File: board-admin-frontend/frontend/lib/features/notifications/model/notification_request.dart
class NotificationRequest {
  final String title;
  final String message;
  final String type;
  final int? createdByUserId;
  final int? targetUserId;
  final int? relatedMeetingId;
  final int? relatedPaperId;
  final int? relatedCommentId;
  final int? relatedAttachmentId;
  final bool announcement;

  const NotificationRequest({
    required this.title,
    required this.message,
    required this.type,
    this.createdByUserId,
    this.targetUserId,
    this.relatedMeetingId,
    this.relatedPaperId,
    this.relatedCommentId,
    this.relatedAttachmentId,
    this.announcement = false,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'message': message,
      'type': type,
      'createdByUserId': createdByUserId,
      'targetUserId': targetUserId,
      'relatedMeetingId': relatedMeetingId,
      'relatedPaperId': relatedPaperId,
      'relatedCommentId': relatedCommentId,
      'relatedAttachmentId': relatedAttachmentId,
      'announcement': announcement,
    };
  }
}

File: board-admin-backend/src/main/java/com/portSrilanka/board_admin_backend/dto/notification/NotificationRequest.java
package com.portSrilanka.board_admin_backend.dto.notification;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class NotificationRequest {
    private String title;
    private String message;
    private String type;
    private Long createdByUserId;
    private Long targetUserId;
    private Long relatedMeetingId;
    private Long relatedPaperId;
    private Long relatedCommentId;
    private Long relatedAttachmentId;
    private boolean announcement;
}
